refund.sol

```diff
@@ -0,0 +1,91 @@
+/*
+
+CODE TO MAKE THE DAO INTO AN AUTOMATIC REFUND
+
+*/
+// A generic token contract
+contract token { 
+    mapping (address => uint256) public balanceOf;  
+    function transferFrom(address _from, address _to, uint256 _value) returns (bool success);
+    function mintToken(address target, uint256 mintedAmount);
+    uint256 public totalSupply;
+}
+
+
+/* define 'owned' */
+contract owned {
+    address public owner;
+
+    function owned() {
+        owner = msg.sender;
+    }
+
+    modifier onlyOwner {
+        if (msg.sender != owner) throw;
+        _
+    }
+
+    function transferOwnership(address newOwner) onlyOwner {
+        owner = newOwner;
+    }
+}
+
+contract DAORefund is owned {
+    // address of original DAO
+    token DAOTokens;
+    
+    // Address of new refund token (can be used to claim any additional eventual refund in the future)
+    token DAOBadgeOfHonor;
+    
+    // the price it will refund it for
+    uint public weiRefundedPerToken;
+
+    /* First time setup */
+    function DAORefund(address DAOAddress, address BOHAddress) {
+        DAOTokens = token(DAOAddress);
+        DAOBadgeOfHonor = token(BOHAddress);
+    }
+    
+    // to be called by the owner to lock the price so it guarantees all tokens are refunded the same
+    function setPrice() onlyOwner {
+        uint tokenSupply = DAOTokens.totalSupply() - DAOTokens.balanceOf(this);
```

Comment:
We will probably have to think a lot about the total supply here. Remember that the exploit allowed for people to create as many tokens as they want into a child DAO, stealing ether in the process and never burning any mother DAO tokens, thus not reducing the total supply of the mother DAO.

What about people who may have already gotten into a Dark DAO and are not malicious? We should probably add some kind of exception for them here.

I think that we should probably do some accounting, see who was naughty, see who may be an innocent victim in a dark DAO and then incorporate that accounting in here.

We should also consider blacklisting all known attackers from any refund. More money for the rest of the legitimate DTH.